helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -210,6 +213,18 @@ private void updateStateTransitionMessageThreadPool(Message message, HelixManage
       return;
     }
 
+    if (!_batchMessageThreadpoolChecked) {
```

Comment:
I see, just to confirm, is backward compatibility the only concern here?
I'm wondering if constructing HelixTaskExecutor with a HelixManager would make more sense, at all, if we're free to change the constructor?